pilot/pkg/model/context.go

```diff
@@ -857,6 +876,11 @@ func (node *Proxy) SupportsIPv6() bool {
 	return node.ipv6Support
 }
 
+// GRPCGen returns true if the "proxy" expects gRPC-compatible xDS responses.
+func (node *Proxy) GRPCGen() bool {
+	return node.Metadata.Generator == "grpc"
```

Comment:
Does grpc gracefully handle the stuff we put in cluster? the mtls (especially auto mtls), custom filters for telemetry, etc?

Reply:
No. That's the biggest concern here – the goal is to attempt to use as many istio configs with gRPC as via integration tests to find the breakages. 

At what point does it become "too many" gRPC only branches in the generator?